src/js/components/MapPanelLocationBar.jsx

```diff
@@ -73,14 +98,18 @@ export default class MapPanelLocationBar extends React.Component {
                 // But we also have the bonus of not needing to make a reverse geocode request
                 // for small changes of the map center.
                 if (delta > MAP_UPDATE_DELTA) {
-                    this.reverseGeocode(currentLatLng);
-                    this.setState({
-                        bookmarkActive: false,
-                        latlng: {
-                            lat: currentLatLng.lat,
-                            lng: currentLatLng.lng
-                        }
-                    });
+                    this.reverseGeocode(currentLatLng)
+                        .then((state = {}) => {
```

Comment:
There's a couple of things that stack up here.

`this.reverseGeocode()` previously had a number of `setState()`s inside of it which occur after a `fetch()`, so it's asynchronous (meaning React won't batch it for us). When followed immediately by another `this.setState()`, this means React needed to render state two to three times per action.

We need to condense this all down to one. Inside of `this.reverseGeocode()` we can assemble all the state-modifying statements by modifying a `state` object and then finally sending that off to `this.setState(state)`. But that's still making two calls to `setState()`, one inside `reverseGeocode()` and one outside of it.

How do you manually batch state updates when happens in an async function and the other does not? Since `reverseGeocode()` uses a Promise (the `fetch`) we can simply return that, with the value of its `state` object, and then combine that with our `state.latlng`.

Back to the original question re: the map, we were already only calling it once, which is just fine. That hasn't changed. But if you're referring to the `constructor()` where I'm setting the intial state to `0`, that's just because there's no map available to look at when this component mounts. So right now it's just sitting behind a `EventEmitter` subscription.

Hope that helps?